fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+`INTERSECTS`; and the temporal operator `ANYINTERACTS`. 
+
+The ANYINTERACTS operator has effectively the same semantics as the existing `datetime` filter.
+
+CQL enables these types of queries:
+- Use of Item Property values in predicates (e.g., `item.properties.eo:cloud_cover`), using comparison operators
+- Items whose `datetime` values are in the month of August of the years 2017-2021, using OR and ANYINTERACTS
+- Items whose `geometry` values intersect any one of several Polygons, using OR and INTERSECTS
+- Items whose `geometry` values intersect one Polygon, but do not intersect another one, using AND, NOT, and
+  INTERSECTS
+
+This extension also supports the Queryables mechanism that allows discovery of what Item fields can be used in 
+predicates.
+
+## OAFeat Part 3 Conformance Classes
+
+OAFeat CQL defines several conformance classes that allow implementers to create arbitrary compositions of 
+functionality that support whatever expressiveness they need.  Implementers many choose not to incur the cost of 
+implementing functionality they do not need, or may not be able to implement functionality that is not supported by their underlying datastore.  For example, Elasticsearch does not support the spatial predicates required by the Enhanced Spatial Operators conformance class.
+
+The Filter extension **requires** support of these three conformance classes:
+
+- Filter (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/filter`) - defines the Queryables mechanism and 
+  parameters `filter-lang`, filter-crs`, and `filter`
+- Features Filter (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/features-filter`) - defines that the parameters defined in `Filter` apply to the Features endpoint (`/collections/{collectionId}/items`) defined by OAFeat Part 1. 
+- Simple CQL (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/simple-cql`) - defines the query language used for the `filter` parameter defined by Filter
+
+This STAC Filter extension extends the Filter conformance class such that these parameters also apply
+to the STAC Item Search resource (/search). The OAFeat Filter conformance class already requires that these
+parameters work for GET requests to the Items resource (/collections/collectionId/items). POST with a JSON body to the Items resource is not supported, as POST is used by the Transactions Extension for creating Item objects.
+
+Additional, the implementation must support at least one of "CQL Text" (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/cql-text`) or "CQL JSON" (`http://www.opengis.net/spec/ogcapi-features-3/1.0/req/cql-json`). If both are advertised as being supported, it is only required that both be supported for GET query parameters, and that only that CQL JSON be supported for POST JSON requests. It is recommended that clients use CQL Text in GET requests and CQL JSON in POST requests. 
```

Comment:
> (more line breaks would be good here to comment more precisely)

I copied the remark linting config from stac-spec the other day, and apparently the 150 line limit is commended out in that config? Anyway, that's why remark wasn't telling me there was a problem, so I'm going to re-enable it and fix all these long lines.

Reply:
You must have looked at a very old version. Best is probably to simply copy/paste https://github.com/radiantearth/stac-spec/blob/master/.circleci/rc.yaml

Reply:
I must have something wrong with my local dev branch. I'm going to recopy it in.

Reply:
> I was also thinking we might make more specific recommendations. Like in item-search filter we require cql-text, in line with our requirement for GET. Also I'm not sure how cql-json is supposed to work for GET? You include JSON in the body? I think we should recommend against that...

cql-json works with GET the same as cql-text does -- the JSON is just a string value to the query parameter.

Reply:
This also indicates we may need our own, more specific conformance classes, since we should presumably allow an implementation to only support CQL Text against the OGC Features endpoint, and only support CQL JSON against the /search endpoint, but right now, there's no way to advertise it as that restricted. I don't think anyone will want to do that, but it's an edge case that reveals how fine-grained the conformance classes need to be.